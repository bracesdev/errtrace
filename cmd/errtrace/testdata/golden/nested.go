//go:build ignore

package foo

import (
	"errors"
	"fmt"
)

func HasFunctionLiteral() {
	err := func() error {
		return errors.New("sadness")
	}()

	fmt.Println(err)
}

func ImmediatelyInvokedFunctionExpression() error {
	return func() error {
		return errors.New("sadness")
	}()
}
