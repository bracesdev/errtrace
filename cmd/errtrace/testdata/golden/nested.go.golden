//go:build ignore

package foo

import (
	"errors"
	"fmt"; "braces.dev/errtrace"
)

func HasFunctionLiteral() {
	err := func() error {
		return errtrace.Wrap(errors.New("sadness"))
	}()

	fmt.Println(err)
}

func ImmediatelyInvokedFunctionExpression() error {
	return errtrace.Wrap(func() error {
		return errtrace.Wrap(errors.New("sadness"))
	}())
}
